java/src/main/native/include/jni_utils.hpp

```diff
@@ -68,6 +69,24 @@ inline void check_java_exception(JNIEnv *const env) {
   }
 }
 
+/**
+ * @brief Helper to convert a pointer to a jlong.
+ *
+ * This is useful when, for instance, converting a cudf::column pointer
+ * to a jlong, for use in JNI.
+ */
+template <typename T> jlong to_jlong(T *ptr) {
+  return reinterpret_cast<jlong>(ptr);
+}
+
+template <typename T> jlong to_jlong(std::unique_ptr<T> &&ptr) {
+  return to_jlong(ptr.release());
+}
+
+template <typename T> jlong to_jlong(std::unique_ptr<T> &ptr) {
+  return to_jlong(std::move(ptr));
+}
```

Comment:
I'm not a fan of these seemingly innocuous methods with very nasty side-effects.  They look a _lot_ like the first form, which simply does a cast, while these are also releasing the memory ownership from underneath the `std::unique_ptr`.  The forms that release the memory should have `release` in their name or some other indicator that this isn't just a casting helper.

Reply:
I'm open to using a better name for these functions. `release_to_jlong()`?

Reply:
That works, or maybe `release_as_jlong()`

Reply:
That's much better. Will post the change shortly.

Reply:
Fixed.